CDP4Common/Types/OrderedItemList.cs

```diff
@@ -574,5 +539,61 @@ private void Add(long sortKey, T item)
             this.sortedItems.Add(sortKey, item);
             this.highestKey = Math.Max(sortKey, this.highestKey);
         }
+
+        /// <summary>
+        /// Validates a value for being null.
+        /// </summary>
+        /// <param name="value">
+        /// The value to validate.
+        /// </param>
+        /// <param name="message">
+        /// The message to pass to an exception if validation not passes.
+        /// </param>
+        /// <exception cref="ArgumentNullException">
+        /// Thrown when <see paramref="value"/> is null.
+        /// </exception>
+        private void ValidateValueForNull(object value, string message = "")
+        {
+            if (value == null)
+            {
+                throw new ArgumentNullException(message);
+            }
+        }
+
+        /// <summary>
+        /// Validates an index to be within an allowed range of the underlying sorted list of items.
+        /// </summary>
+        /// <param name="index">
+        /// The value to validate.
+        /// </param>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// Thrown when <see paramref="index"/> is out of the allowed range.
+        /// </exception>
+        private void ValidateIndex(int index)
+        {
+            if (index < 0 || index >= this.sortedItems.Count)
+            {
+                throw new ArgumentOutOfRangeException(
+                    "index", string.Format("The index {0} does not exist in the ordered item list, valid range is 0 to {1}", index, this.sortedItems.Count - 1));
+            }
+        }
+
+        /// <summary>
+        /// Validates whether an item is already in the underlying sorted list of items.
+        /// </summary>
+        /// <param name="item">
+        /// The item to validate.
+        /// </param>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown when <see paramref="item"/> is already in the list.
+        /// </exception>
+        private void ValidateItemForExistence(T item)
+        {
+            if (this.sortedItems.Values.Contains(item))
+            {
+                var thing = item as Thing;
+                throw new InvalidOperationException(string.Format("The sorted list already contains the item {0}", thing != null ? thing.Iid.ToString() : "An item is not a Thing. Incorrect use of validation."));

```

Comment:
please make use of string interpolation **$** syntax instead of **string.format**

Since the class is called OrderedItemList I propose that in the exception message we replace **sorted** with **ordered**